openquake/hazardlib/correlation.py

```diff
@@ -76,20 +87,37 @@ def apply_correlation(self, sites, imt, residuals):
         # of N random numbers (where N is equal to number of sites).
         # we need to do that multiplication once per realization
         # with the same matrix and different vectors.
+
+        # Reshape 'stddev_intra' if needed.
+        stddev_intra_shape = stddev_intra.shape
+        try:
+            if stddev_intra_shape[0] > stddev_intra_shape[1]:
+                stddev_intra = numpy.transpose(stddev_intra)
+                stddev_intra = stddev_intra[0]
+        except:
+            stddev_intra = stddev_intra
```

Comment:
What kind of exception do you expect here?

Reply:
IndexError, if stddev_intra has shape (N,), then stddev_intra_shape[1] throws an IndexError exception, then stddev_intra should not be transpose.
I'll include the IndexError exception.